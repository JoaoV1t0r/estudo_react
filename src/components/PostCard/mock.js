export const postCardMock = {
  title: "Title",
  body: "Body",
  id: 1,
  cover: "img/img.png",
};
