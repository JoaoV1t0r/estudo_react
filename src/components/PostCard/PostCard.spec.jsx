import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardMock } from "./mock";

const props = postCardMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: /title/i })).toHaveAttribute(
      "src",
      props.cover
    );

    expect(screen.getByRole("heading", { name: "Title" })).toBeInTheDocument();

    expect(screen.getByText("Body")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
