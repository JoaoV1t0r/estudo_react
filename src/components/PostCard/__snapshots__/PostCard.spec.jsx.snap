// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <div
    class="post-content"
  >
    <img
      alt="Title"
      src="img/img.png"
    />
    <h2>
      Title
    </h2>
    <p>
      Body
    </p>
  </div>
</div>
`;
